### PR TITLE
[CSL-2122] Allow to specify def tx inputs selection policy in config

### PR DESCRIPTION
**Pull request by Martoon-00**

### CODE REVIEW

File: node/src/Pos/Configuration.hs [hunk 1]
```diff
@@ -76,11 +80,27 @@ data NodeConfiguration = NodeConfiguration
...
     , ccPendingTxResubmissionPeriod :: !Int
       -- ^ Minimal delay between pending transactions resubmission
+    , ccTxDefInputSelectionPolicy   :: !InputSelectionPolicyConf
+      -- ^ Default policy to pick inputs when forming transaction
     } deriving (Show, Generic)
 
 instance FromJSON NodeConfiguration where
     parseJSON = genericParseJSON defaultOptions
 
+-- | Specifies the way Uxtos are going to be grouped.
```
Comment (gromakovsky) [inline node/src/Pos/Configuration.hs]: Typo: Utxos.

Comment (Martoon-00) [reply node/src/Pos/Configuration.hs]: Right, thanks

**Comment by Martoon-00** [comment]
We can't reuse `InputSelectionPolicy` as is, because it deserializes from `{ "groupingPolicy": "<value>" }`, and for config we need to deserialize from plain `"<value>"` 🙂

**Comment by adinapoli-iohk** [comment]
> We can't reuse InputSelectionPolicy as is

Ah, I see, that's because we deserialise from the yaml file which uses the same `Aeson` instances, and thus we need an extra layer of indirection for the config :)

File: node/src/Pos/Client/Txp/Util.hs [hunk 2]
```diff
@@ -147,7 +150,7 @@ isCheckedTxError = \case
 -- Tx creation
 -----------------------------------------------------------------------------
 
--- | Specifies the way Uxtos are going to be grouped.
+-- | Specifies the way Uxto is going to be grouped.
```
Comment (gromakovsky) [inline node/src/Pos/Client/Txp/Util.hs]: There's still a typo :/

Comment (gromakovsky) [reply node/src/Pos/Client/Txp/Util.hs]: Fixed by myself, because Konstantin is on vacation.

**Comment by Martoon-00** [comment]
@adinapoli-iohk Since it's not getting merged, can I close this PR?

**Comment by adinapoli-iohk** [comment]
@Martoon-00 It all boils down to whether or not we have a nice migration path for exchanges using the old API vs exchanges using the new (if you remember, the JSON payload for `newPayment` changed a little bit).

Would you do a bit of investigation and get back to me? My gut feeling is that we might need to rebase this against `master`, review it and make sure it's actually buying us something.

Let's start with baby step first: would you close this PR and open a new one against `master`, describing what this would buy us, and what are the consequences if we don't merge it?
Then we can kickoff the discussion and decide once for all if we want to merge it or not.

TL;DR Close this PR, open a new one, let's discuss 😉

**Comment by Martoon-00** [comment]
Rebased to master, see ^

**Comment by Martoon-00** [comment]
Now we can assess which work exactly should be done :eyes:
